activesupport/lib/active_support/inflector/inflections.rb

```diff
@@ -76,8 +76,7 @@ def self.instance_or_fallback(locale)
       attr_reader :acronyms_camelize_regex, :acronyms_underscore_regex # :nodoc:
 
       def initialize
-        @plurals, @singulars, @uncountables, @humans, @acronyms = [], [], Uncountables.new, [], {}
-        define_acronym_regex_patterns
+        set_defaults
```

Comment:
Maybe we could remove some duplication and call
```suggestion
        clear(:all)
```

And implement clear as follows:
```ruby
def clear(scope = :all)
  case scope
  when :all
    clear(:acronyms)
    clear(:plurals)
    clear(:singulars)
    clear(:uncountables)
    clear(:humans)
  when :acronyms
    @acronyms = {}
    define_acronym_regex_patterns
  when :uncountables,
     @uncountables = Uncountables.new
  when :plurals, :singulars, :humans
    instance_variable_set "@#{scope}", []
  end
end
```

Reply:
It does feel a bit strange to call `#clear` in `#initialize`, but I do like that it deduplicates the knowledge of what types are required for each field.

Reply:
Another solution could be assigning in the initializer but calling `clear` on the Arrays and Hashes in the clear `method`.
This makes it so `clear` doesn't need to know if it's a Hash or Array (and Uncountable?).
```
when :plurals, :singulars, :humans, :uncountables, :acronyms
  instance_variable_get("@#{scope}").clear
```

Reply:
This is how it's done in the routes:
https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/actionpack/lib/action_dispatch/journey/routes.rb#L37-L41